google/cloud/firestore_v1/base_vector_query.py

```diff
@@ -124,5 +134,7 @@ def stream(
         transaction=None,
         retry: retries.Retry = gapic_v1.method.DEFAULT,
         timeout: float = None,
-    ) -> Iterable[document.DocumentSnapshot]:
+        *,
+        explain_options: Optional[ExplainOptions] = None,
+    ) -> Generator[Optional[DocumentSnapshot], Any, Optional[ExplainMetrics]]:
```

Comment:
why Optional[DocumentSnapshot]?

Reply:
I added the `Optional` to cover the case when the generator generates 0 items. Maybe it's not necessary?

Reply:
I think it's not necessary. I removed all similar occurrences.

Reply:
Yeah, that's not necessary. Optional implies it would sometimes `yield None`, which would be different from just not yielding anything